node/src/types/deploy.rs

```diff
@@ -474,16 +477,33 @@ impl Deploy {
         &self.approvals
     }
 
-    /// Returns the header() wrapped the correct DeployType given the `session`.
+    /// Returns the DeployType wrapped the correct DeployType.
     pub fn deploy_type(&self) -> DeployType {
-        match self.session() {
-            ExecutableDeployItem::Transfer { .. } => DeployType::Transfer(self.header().clone()),
-            ExecutableDeployItem::ModuleBytes { .. }
-            | ExecutableDeployItem::StoredContractByHash { .. }
-            | ExecutableDeployItem::StoredContractByName { .. }
-            | ExecutableDeployItem::StoredVersionedContractByHash { .. }
-            | ExecutableDeployItem::StoredVersionedContractByName { .. } => {
-                DeployType::Wasm(self.header().clone())
+        let header = self.header().clone();
+        let size = self.serialized_length();
+        if self.session().is_transfer() {
+            // TODO: non-zero value
+            let payment_amount = Motes::zero();
+            DeployType::Transfer {
+                header,
+                payment_amount,
+                size,
+            }
+        } else {
+            let payment_item = self.payment().clone();
+            let payment_amount = {
+                // Some assumptions here:
+                // Since we are a Transfer variant, we expect: args to exist, contain "amount", and
+                // be a valid U512 value.
+                let args = payment_item.into_runtime_args().expect("should get");
+                let value = args.get(ARG_AMOUNT).expect("should exist");
+                let value = value.clone().into_t::<U512>().expect("should be U512");
```

Comment:
These two `expects` are not something a user can make fail, by putting e.g. a non-number or nothing into the "amount" argument, are they?

Reply:
It's a good point - I've mapped these to a new `deploy::Error` variant, and bubble that up to the validator, log it on on `error!` and returning `Effects::new()` from the event handler if this were to happen.